dart/lib/src/hub.dart

```diff
@@ -0,0 +1,379 @@
+        SeverityLevel.warning,
+        'captureEvent called with null parameter.',
+      );
+    } else {
+      final item = _stack.last;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureEvent(event: event);
+        } catch (err) {
+          /* FIXME(rxlabz) ? Event.id ?*/
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing event with id: ${event}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureEvent',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  @override
+  Future<SentryId> captureException({
+    dynamic throwable,
+    dynamic stackTrace,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureException' call is a no-op.",
+      );
+    } else if (throwable == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureException called with null parameter.',
+      );
+    } else {
+      final item = _stack.last;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureException(
+            throwable,
+            stackTrace: stackTrace,
+          );
+        } catch (err) {
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing exception : ${throwable}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureException',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  @override
+  Future<SentryId> captureMessage(
+    Message message, {
+    SeverityLevel level,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureMessage' call is a no-op.",
+      );
+    } else if (message == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureMessage called with null parameter.',
+      );
+    } else {
+      final item = _stack.last;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureMessage(
+            message: message,
+            level: level,
+          );
+        } catch (err) {
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing message with id: ${message}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureMessage',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  @override
+  void addBreadcrumb(Breadcrumb breadCrumb) {
+    // TODO: implement addBreadcrumb
+    throw UnimplementedError();
+  }
```

Comment:
we can skip it, ideally people will do that thru `configureScope`, the reason is, we've added them on Java to easier migration from Java SDK v1 to v2, but now we are not gonna add them